### PR TITLE
Feature/851

### PR DESCRIPTION
**Pull request by henr9533**
# Description

A new folder containing custom exceptions has been added, which currently contains three different exceptions. They are however similar and can later be changed to be much more unique if so required. Furthermore, multiple try-catch statements have been implemented in various areas of the code, to ensure a stable launch.

Fixes #\851

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

A new test file has been created which includes three tests to make sure that the new custom exceptions exist and are thrown correctly.

**Development Configuration**

* Flutter version: 3.3.8
* Dart version: 2.18.4

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [x] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -22,7 +22,7 @@ void main() {
    * Use the "environments.local.json" for running against your local web-api
    * For IOS users: change the SERVER_HOST in the environment.local file to "http://localhost:5000"
    */
-  environment.setFile('assets/environments.dev.json').whenComplete(() {
+  environment.setFile('assets/environments.local.json').whenComplete(() {
```
Comment (JJTM20) [inline lib/main.dart]: Shouldn't this be changed back to `environment.setFile('assets/environments.dev.json')`?

**Review by JJTM20 — CHANGES_REQUESTED** [review]
1 comment, the rest looks fine:)

File: lib/main.dart (hunk 1 above)
Comment (ghost) [inline lib/main.dart]: Dette skal forblive `dev`, right?

File: lib/screens/login_screen.dart [hunk 2]
```diff
@@ -61,29 +64,40 @@ class LoginScreenState extends State<LoginScreen> {
...
-              }
-            }).catchError((Object error) {
-              unknownErrorDialog(error.toString());
-            });
-          } else {
-            creatingNotifyDialog(
-                'Der er ingen forbindelse'
-                    ' til internettet.',
-                'NoConnectionToInternet');
+          //Er ikke sikker på den her try-catch nogen sinde "catcher"
```
Comment (ghost) [inline lib/screens/login_screen.dart]: I guess this comment should be in English

File: test/exceptions/custom_exceptions_test.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  void throwEditWeekplanButtonException() {
+    throw EditWeekplanButtonException('Test', 'Test');
+  }
+
+  void throwOrientationException() {
+    throw OrientationException('Test', 'Test');
+  }
+
+  test('Should throw ServerException', () {
+    expect(throwServerException, throwsA(isA<ServerException>()));
```
Comment (ghost) [inline test/exceptions/custom_exceptions_test.dart:18]: Hvad er det lige `throwsA` og `isA` gør? 😅

File: test/widgets/pictogram_image_test.dart [hunk 4]
```diff
@@ -146,22 +146,19 @@ void main() {
     await done.future;
     await tester.pumpWidget(PictogramImage(
       pictogram: pictogramModel,
-      onPressed: () {  },
+      onPressed: () {},
```
Comment (ghost) [inline test/widgets/pictogram_image_test.dart:149]: Behøver `onPressed` at være der?

**Review by ghost — COMMENTED** [review]
Looks good, har bare lige et par spørgsmål

File: lib/main.dart (hunk 1 above)
Comment (henr9533) [reply lib/main.dart]: Har du ret i. Den er blevet fikset :)

Comment (henr9533) [reply lib/main.dart]: Yes, den er lige blevet ændret tilbage :)

File: lib/screens/login_screen.dart (hunk 2 above)
Comment (henr9533) [reply lib/screens/login_screen.dart]: Rettet til engelsk

File: test/exceptions/custom_exceptions_test.dart (hunk 3 above)
Comment (henr9533) [reply test/exceptions/custom_exceptions_test.dart:18]: throwsA er en funktion, som tjekker om der bliver smidt en specifik exception hvorimod isA definerer hvilken type af exception det er 😃

File: test/widgets/pictogram_image_test.dart (hunk 4 above)
Comment (henr9533) [reply test/widgets/pictogram_image_test.dart:149]: Den er der fra develop branch af, så har ikke ændret i den. Bare automatisk formatting der er blevet lavet 🙂

**Review by YusufIA99 — DISMISSED** [review]
Looks good to me 👍

**Review by JJTM20 — DISMISSED** [review]
lgtm+

**Review by YusufIA99 — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]
Sgutm

**Review by karlbarnholdt — APPROVED** [review]
Super

**Review by J2KJonas — APPROVED** [review]

**Pull request merged**